lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3329,7 +3329,17 @@ private function isIdentifierEquals($entity1, $entity2)
             ? $this->entityIdentifiers[$oid2]
             : $this->identifierFlattener->flattenIdentifier($class, $class->getIdentifierValues($entity2));
 
-        return $id1 === $id2 || implode(' ', $id1) === implode(' ', $id2);
+        return $id1 === $id2 || $this->hashIdentifier($id1) === $this->hashIdentifier($id2);
+    }
+
+    /**
+     * @param array $identifier
+     *
+     * @return string
+     */
+    private function hashIdentifier(array $identifier)
+    {
+        return serialize(array_values(array_map('strval', $identifier)));
```

Comment:
I mean lets take a real world example. Lets say we have a website that takes 100 entities (which is quite a lot already) and that needs to render. What impact (absolute) does that have? Like 20ms? Who will recognize that? Nobody. Everything that goes beyond 100 entries at a time should not be done with ORM IMO. Of course if my stupidly guessed 20ms is in fact 2 seconds, it's a different story but otherwise who cares?